### PR TITLE
feat: add `Seek()` method to `Decoder`

### PR DESCRIPTION
**Pull request by dylan-bourque**
add new `Seek()` method (satisfying `io.Seeker`) to `Decoder` with associated tests
also add `Mode()` to expose the read mode (fast vs safe) to consumers

### CODE REVIEW

File: decoder.go [hunk 1]
```diff
@@ -66,11 +66,44 @@ func NewDecoder(p []byte) *Decoder {
 	}
 }
 
+// Mode returns the current decoding mode, safe vs fastest.
```
Comment (wmorgan6796) [inline decoder.go:69]: can you update the PR title/body to mention this change as well.

**Review by wmorgan6796 — APPROVED** [review]
Looks good, only request is that you update the PR title or body to mention the adding of the `Mode()` function

File: decoder.go (hunk 1 above)
Comment (dylan-bourque) [reply decoder.go:69]: It's mentioned in the PR description and the body of the commit message

Comment (wmorgan6796) [reply decoder.go:69]: My ability to read appears to have regressed in the new year

**Pull request merged**